editor/src/main/com/mbrlabs/mundus/editor/input/FreeCamController.kt

```diff
@@ -166,4 +171,33 @@ class FreeCamController : InputAdapter() {
         }
         camera!!.update(true)
     }
+
+    override fun mouseMoved(screenX: Int, screenY: Int): Boolean {
+        if (camera == null) {
+            return false
+        }
+
+        val currentProject = projectManager.current()
+        val currentScene = currentProject.currScene
+
+        val ray = currentScene.viewport.getPickRay(screenX.toFloat(), screenY.toFloat())
+
+        var helperCell: HelperLineCenterObject? = null
+
+        for (terrain in currentScene.terrains) {
+            val result = getRayIntersection(terrain, ray, Vector3())
```

Comment:
Do we need to do all this calculation logic on mouse move when helper lines are disabled?

Reply:
Thank you for your review. I rewrite that codes to use GameObjectPicker and added getRayIntersection method to TerrainComponent what calls Terrain's getRayIntersection method.